mindsdb/interfaces/datastore/datastore.py

```diff
@@ -104,18 +102,12 @@ def delete_datasource(self, name):
             pass
 
     def save_datasource(self, name, source_type, source, file_path=None):
-        datasource_record = Datasource(company_id=self.company_id, name=name)
-
         if source_type == 'file' and (file_path is None):
             raise Exception('`file_path` argument required when source_type == "file"')
 
         ds_meta_dir = os.path.join(self.dir, name)
         os.mkdir(ds_meta_dir)
 
-        session.add(datasource_record)
-        session.commit()
-        datasource_record = session.query(Datasource).filter_by(company_id=self.company_id, name=name).first()
```

Comment:
In principle I agree with only creating the datasource in a delayed manner in the database, the issue here will be one of locking, i.e. atm we have to mechanism to "lock" the ability for multiple clients to work on an entitiy (e.g. datasource) with the same name, unless it's already in the database, so this could cause issues.

But, regardless, I think it's unlikely to happen. So keep this as it.

Still, please don't delete "broken" datasources, keep them in the database so we can figure out why they are broken later.